src/systems/systems/src/pressurization/mod.rs

```diff
@@ -106,6 +224,114 @@ impl Default for Pressurization {
+    pub fn new() -> Self {
+        Self {
+            mode_sel: AutoManFaultPushButton::new_auto("PRESS_MODE_SEL"),
+            man_vs_ctl_switch: SpringLoadedSwitch::new("PRESS_MAN_VS_CTL"),
+            ldg_elev_knob: ValueKnob::new_with_value("PRESS_LDG_ELEV", -2000.),
+            ditching: NormalOnPushButton::new_normal("PRESS_DITCHING"),
+        }
+    }
+
+    fn ldg_elev_is_auto(&self) -> bool {
+        let margin = 100.;
+        (self.ldg_elev_knob.value() + 2000.).abs() < margin
+    }
+
+    fn is_in_man_mode(&self) -> bool {
+        !self.mode_sel.is_auto()
+    }
+
+    fn man_vs_switch_position(&self) -> usize {
+        self.man_vs_ctl_switch.toggle_position()
+    }
+}
+
+impl SimulationElement for PressurizationOverheadPanel {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.mode_sel.accept(visitor);
+        self.man_vs_ctl_switch.accept(visitor);
+        self.ldg_elev_knob.accept(visitor);
+        self.ditching.accept(visitor);
+
+        visitor.visit(self);
+    }
+}
+impl Default for PressurizationOverheadPanel {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ControllerSignal<PressureValveSignal> for PressurizationOverheadPanel {
+    fn signal(&self) -> Option<PressureValveSignal> {
+        if !self.is_in_man_mode() {
+            None
+        } else {
+            match self.man_vs_switch_position() {
+                0 => Some(PressureValveSignal::Open),
+                2 => Some(PressureValveSignal::Close),
+                _ => Some(PressureValveSignal::Neutral),
+            }
+        }
+    }
+}
+
+struct ResidualPressureController {
+    should_open_outflow_valve: bool,
+    timer: Duration,
+}
+
+impl ResidualPressureController {
+    fn new() -> Self {
+        Self {
+            should_open_outflow_valve: false,
+            timer: Duration::from_secs(0),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+        outflow_valve_open_amount: Ratio,
+        is_in_man_mode: bool,
+        lgciu_gear_compressed: bool,
+    ) {
+        if outflow_valve_open_amount < Ratio::new::<percent>(100.)
+            && is_in_man_mode
+            && lgciu_gear_compressed
+            && (!(engines
+                .iter()
+                .any(|&x| x.corrected_n1() > Ratio::new::<percent>(15.)))
+                || context.indicated_airspeed() < Velocity::new::<knot>(70.))
+        {
+            self.timer += context.delta();
+        } else {
+            self.timer = Duration::from_secs(0);
+        }
+        self.should_open_outflow_valve = self.timer > Duration::from_secs(60);
+    }
+}
+
+impl ControllerSignal<PressureValveSignal> for ResidualPressureController {
+    fn signal(&self) -> Option<PressureValveSignal> {
+        if self.should_open_outflow_valve {
+            Some(PressureValveSignal::Open)
+        } else {
+            None
+        }
+    }
```

Comment:
**suggestion**: As `self.should_open_outflow_valve` is only read once in the `signal` code, consider moving the expression into the `signal` function and removing the field.

You could also consider adding a function `should_open_outflow_valve()` to make it more expressive.
```rust
    fn should_open_outflow_valve(&self) -> bool {
        self.timer > Duration::from_secs(60)
    }

    fn signal(&self) -> Option<PressureValveSignal> {
        if self.should_open_outflow_valve() {
            Some(PressureValveSignal::Open)
        } else {
            None
        }
    }
```

Reply:
Good spot, I moved the logic directly to the `signal` code.